lib/pacemaker/pcmk_sched_colocation.c

```diff
@@ -594,24 +596,25 @@ unpack_simple_colocation(xmlNode *xml_obj, const char *id,
                                                  XML_COLOC_ATTR_TARGET_ROLE);
     const char *attr = crm_element_value(xml_obj, XML_COLOC_ATTR_NODE_ATTR);
 
-    // @COMPAT: Deprecated since 2.1.5
-    const char *dependent_instance = crm_element_value(xml_obj,
-                                                       XML_COLOC_ATTR_SOURCE_INSTANCE);
-    // @COMPAT: Deprecated since 2.1.5
-    const char *primary_instance = crm_element_value(xml_obj,
-                                                     XML_COLOC_ATTR_TARGET_INSTANCE);
+    const char *primary_instance = NULL;
+    const char *dependent_instance = NULL;
+    pe_resource_t *dependent = NULL;
+    pe_resource_t *primary = NULL;
```

Comment:
Complete nitpick: I'd swap order just to be consistent (it's primary then dependent in the other uses)

Reply:
might as well